### PR TITLE
fix various Travis issues (update CMake, fail on error, boost and ACE)

### LINKED ISSUE #174: ACE fails to build on Travis trusty
[Example log](https://api.travis-ci.org/v3/job/465686977/log.txt) (with -j 2 unfortunately). ACE building seems to be ok but ends with
```
ACE_Qt4Reactor_moc will not be built due to the following disabled make macros: qt4
GNUmakefile: /home/travis/build/CCPPETMR/SIRF-SuperBuild/builds/ACE/build/source/ace/QtReactor/GNUmakefile.ACE_Qt4Reactor_moc MAKEFLAGS=s

make[5]: *** [stamp/ACE-build] Error 141
```

### PR DESCRIPTION
**Pull request by paskino**
installs CMake 3.13.1
calls travis_terminate 1 in the whole block running cmake config and make (was only at make)

closes #174 (possibly)

### CODE REVIEW

**Comment by paskino** [comment]
Do you know why this [`export PATH="$PWD/cmake-*/bin:$PATH"`](https://github.com/CCPPETMR/SIRF-SuperBuild/blob/travis_cmake/.travis.yml#L199) doesn't seem to work?

**Comment by KrisThielemans** [comment]
i see before that
```
gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```
therefore something wrong with the download

**Comment by KrisThielemans** [comment]
I've added a verbose to curl.

This is WIP, with likely many dumb commits. We will squash merge this likely) (so therefore: do not merge this PR/branch elsewhere)

**Comment by KrisThielemans** [comment]
The boost problem is known. Trusty is too old. so we need to set `USE_SYSTEM_BOOST=OFF` in all linux cases

**Comment by KrisThielemans** [comment]
CMake version fixed now (will need to squash those commits!). But ACE still fails cryptically
```
make[7]: warning: jobserver unavailable: using -j1.  Add `+' to parent make rule.
make[7]: warning: jobserver unavailable: using -j1.  Add `+' to parent make rule.
GNUmakefile: /home/travis/build/CCPPETMR/SIRF-SuperBuild/builds/ACE/build/source/ace/GNUmakefile.ACE MAKEFLAGS=s
make[7]: warning: jobserver unavailable: using -j1.  Add `+' to parent make rule.
ACE_Qt4Reactor_moc will not be built due to the following disabled make macros: qt4
GNUmakefile: /home/travis/build/CCPPETMR/SIRF-SuperBuild/builds/ACE/build/source/ace/QtReactor/GNUmakefile.ACE_Qt4Reactor_moc MAKEFLAGS=s
make[5]: *** [stamp/ACE-build] Error 141
```

**Comment by KrisThielemans** [comment]
I've squashed the commit series to have one "fix cmake". To do:
- change `USE_SYSTEM_BOOST=OFF` due to Travis trusty too old
- fix ACE build.

I feel that for the latter, we can use a separate PR, as we don't need it on the VM nor bionic. So I suggest to put `USE_SYSTEM_ACE=ON` for now (with a comment why), and create an issue for later. Hopefully we'll be green then.

**Comment by paskino** [comment]
I've got no clue on why travis doesn't run. @casperdcl @KrisThielemans @bathomas  any ideas?

**Comment by KrisThielemans** [comment]
https://travis-ci.org/CCPPETMR/SIRF-SuperBuild/requests says "could not parse .travis.yml" for your last 2 commits.I found http://lint.travis-ci.org but it says it no longer works but refers to a WIP that seems very complicated to get to run. I'm now looking with the [command line version](https://support.travis-ci.com/hc/en-us/articles/115002904174-Validating-travis-yml-files). (Note: needs `ruby-dev` for installation)

**Comment by KrisThielemans** [comment]
running now. I've cancelled the PR build but kept the [branch build](https://travis-ci.org/CCPPETMR/SIRF-SuperBuild/builds/467384823) to save some global warming. (Seems to me that if we have a PR from our own branch, Travis runs everything twice)

**Comment by paskino** [comment]
I thought I had fixed those spaces.

It was late

**Comment by KrisThielemans** [comment]
> Oh yes, it's late and I misunderstood the convention.
it must have been even later when you tried to fix it :-)
```
   # +boost -itk -fftw3 -hdf5 -ace +niftyreg
   env: EXTRA_BUILD_FLAGS="-DUSE_SYSTEM_Boost=OFF -DUSE_ITK=OFF -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DUSE_SYSTEM_ACE=ON -DUSE_NiftyReg=ON" MATRIX_EVAL="CC=gcc-5 CXX=g++-5" PYMVER=3
```
should be
```
   # -boost -itk +fftw3 +hdf5 +ace +niftyreg
```

**Comment by KrisThielemans** [comment]
let's hope this works. If it does, I suggest you fix the comments and squash the last few commits (fixing commit message by starting with `[TRAVIS]` and saying what you did with ACE as well), then we re-enable OSX. Let me know if you need any help with this.

**Comment by paskino** [comment]
> ```
>    # +boost -itk -fftw3 -hdf5 -ace +niftyreg
>    env: EXTRA_BUILD_FLAGS="-DUSE_SYSTEM_Boost=OFF -DUSE_ITK=OFF -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON -DUSE_SYSTEM_ACE=ON -DUSE_NiftyReg=ON" MATRIX_EVAL="CC=gcc-5 CXX=g++-5" PYMVER=3
> ```
> should be
> 
> ```
>    # -boost -itk +fftw3 +hdf5 +ace +niftyreg
> ```

You mean that it refers to the fact that the field is ON (+) or OFF (-) regardless of the meaning that may be reversed. Personally I find it very confusing. It'd be more rational to have that comment to highlight which packages get built or not. The other info can be gathered from the `env` line.

**Comment by KrisThielemans** [comment]
sure. but there are some other `ON` things in there, where the + does make more sense.

anyway, consistency between comments and code is important. sticking to Casper's will mean minimal .travis.yml changes, so I'd do that (and probably add a comment at the top that + means `ON`)

File: .travis.yml [hunk 1]
```diff
@@ -13,45 +13,49 @@ language: cpp
...
  - os: linux
    python: 3
-   # +DEVEL +boost -hdf5 -fftw3 +siemens_to_ismrmrd
-   env: EXTRA_BUILD_FLAGS="-DDEVEL_BUILD=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_HDF5=OFF -DUSE_SYSTEM_FFTW3=OFF -DBUILD_siemens_to_ismrmrd=ON" MATRIX_EVAL="CC=gcc-6 CXX=g++-6" PYMVER=3
+   # +DEVEL -boost -hdf5 -fftw3 +siemens_to_ismrmrd
+   env: EXTRA_BUILD_FLAGS="-DDEVEL_BUILD=ON -DUSE_SYSTEM_Boost=OFF -DUSE_SYSTEM_HDF5=OFF -DUSE_SYSTEM_FFTW3=OFF -DBUILD_siemens_to_ismrmrd=ON" MATRIX_EVAL="CC=gcc-6 CXX=g++-6" PYMVER=3
  - os: linux
    python: 2.7
-   # +DEVEL -boost -fftw3 -hdf5 -swig
+   # +DEVEL -boost -fftw3 +hdf5 -swig
```
Comment (KrisThielemans) [inline .travis.yml]: -hdf5

**Comment by KrisThielemans** [comment]
[one job](https://travis-ci.org/CCPPETMR/SIRF-SuperBuild/jobs/467486510) failed with
```
WARNING: The following packages cannot be authenticated!
  gcc-8-base libgcc1 libstdc++6 gcc-5-base libasan2 gcc-6-base libasan3
  libatomic1 libcc1-0 gcc-7-base libcilkrts5 libgomp1 libquadmath0
  libgfortran3 libisl15 libitm1 liblsan0 libmpfr4 libmpx0 libmpx2 libtsan0
  libubsan0 cpp-5 cpp-6 libgcc-5-dev gcc-5 libstdc++-5-dev g++-5 libgcc-6-dev
  gcc-6 libstdc++-6-dev g++-6
E: There were unauthenticated packages and -y was used without --allow-unauthenticated
apt-get.diagnostics
apt-get install failed
....
W: GPG error: http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu trusty InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1E9377A2BA9EF27F
W: The repository 'http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu trusty InRelease' is not signed.
W: There is no public key available for the following key IDs:
1E9377A2BA9EF27F  
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends $(travis_apt_get_options) install git-core build-essential g++-5 g++-6 libboost-all-dev libhdf5-serial-dev libfftw3-dev python-dev python3-dev python-tk python3-tk libopenblas-dev libatlas-base-dev liblapack-dev libxml2-dev libarmadillo-dev libgtest-dev libplplot-dev swig ccache realpath libxslt-dev libace-dev" failed and exited with 100 during .
```
that isn't us obviously. I guess I'd ignore it.

File: .travis.yml (hunk 1 above)
Comment (paskino) [reply .travis.yml]: damn! I'll wait for travis to finish and then correct it.

**Comment by paskino** [comment]
@KrisThielemans I think it can be merged now.

**Comment by KrisThielemans** [comment]
There was one more boost left over. I've also squashed your 2 commits into 1 and fixed the log message. Let's see now.

**Comment by paskino** [comment]
OK, I had not seen your force push @KrisThielemans , which was like mine + a squash, so I force pushed deleting your commit. Now, is it possible to go backward?
Or let's keep it like this?

**Comment by KrisThielemans** [comment]
yeah... that's the problem with force push of course. I didn't think you'd still be working on this. sorry.

also interesting to note that this made "my" Travis jobs fails, as the commit no longer existed. (makes sense of course).

I'm afraid I like my log message more :-; (in particular, I feel we need `[TRAVIS]` in the first line, but not in the detail of the log message) So I'll force push again if that's ok.

**Comment by KrisThielemans** [comment]
I'll go ahead with the push.

**Comment by paskino** [comment]
Please do.

I'm committed....

**Pull request merged**